docs/docs/how-tos/configure-gitops.md

```diff
@@ -0,0 +1,75 @@
+---
+id: configure-gitops
+title: Setup Nebari GitOps
+description: Setting up GitOps for Nebari
+---
+
+# Setup Nebari gitops
+
+## What is GitOps / Infrastructure as Code (IaC)?
+
+GitOps also known as Infrastructure as Code(Iac) is the practice of using git for
+managing the state of your infrastructure. This allows for easy
+auditing and discovering the current state of your system.
+
+## Setting up IaC on Github-Actions
+
+Configuring Neabri to render the appropriate workflows for
+github-actions can be configured via the following settings in your
+`nebari-config.yaml`.
+
+```yaml
+ci_cd:
+  type: github-actions
+  branch: main
+  commit_render: true
```

Comment:
I've read the whole doc and I still don't know how this works.
* after I put this in my config, what do I need to run?
* if I include this here, then run `nebari deploy` it will generate some `/.github/workflow/deploy.yml` scripts for me?
* What do the "appropriate workflows" look like?
* Will I need to commit them to a repo?
* When I make changes to the repo, what will happen? Is it watching all files, or only certain files?
* What step am I automating here?

Reply:
I just added gitops to an existing Nebari deployment that was not set up initially with that functionality. We did not use the guided set-up - we deployed Nebari using a configuration file. It may be important to note that we did include this in the configuration yaml file during initial deployment (I'm not sure if a user who did not include this would have to take additional steps before the steps I took):
```
ci_cd:
  type: github-actions
  branch: main
  commit_render: true
```
To set up gitops, I then
1. created a repo and pushed all files from deployment to it (`.github`, `stages`, `.gitignore`, `nebari-config.yaml`).
2. Created the following secrets for Github Actions to use: `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `REPOSITORY_ACCESS_TOKEN` with the keys/tokens that I used to deploy Nebari.
3. I enabled ACLs on the S3 bucket created by my Nebari deployment (named `{project_name}-dev-terraform-state`. I used my AWS console to find the bucket, clicked on the Permissions Tab, scrolled to Object Ownership and clicked edit, then selected ACLs enabled. (see [here](https://stackoverflow.com/questions/70333681/for-an-amazon-s3-bucket-deployment-from-github-how-do-i-fix-the-error-accesscont)).

After taking these steps, I was able to make a change to my configuration file and open a PR. This triggered the `github/workflows/nebari-linter.yaml` job. When that passed, I merged the PR, which triggered the job created by `.github/workflows/nebari-ops.yaml` and the update to our deployment was successful.